appscale/tools/agents/azure_agent.py

```diff
@@ -246,7 +253,7 @@ def describe_instances(self, parameters, pending=False):
       instance_ids: A list of unique Azure VM names.
     """
     credentials = self.open_connection(parameters)
-    subscription_id = parameters[self.PARAM_SUBSCRIBER_ID]
+    subscription_id = str(parameters[self.PARAM_SUBSCRIBER_ID])
```

Comment:
It shouldn't be but somehow there is a problem with the Azure SDK, which unless I specifically convert it to str, it always complains that 'Subscription ID should be only be a string'. I put that check in only when I started facing that issue. Its weird and I was not able to find a reason why.